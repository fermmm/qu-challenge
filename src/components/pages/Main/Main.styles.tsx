// tslint:disable: typedef
import styled from 'styled-components';

export const MainContainer = styled('div')`
   display: flex;
   align-items: center;
   justify-content: center;
`;
