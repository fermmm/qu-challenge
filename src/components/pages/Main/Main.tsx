import React, { FC, useEffect } from 'react';
import { hot } from 'react-hot-loader';
import { useFilms } from '../../../api/films';
import { Column } from '../../common/Column/Column';
import { FilmsList } from '../FilmsList/FilmsList';

export const Main: FC = () => {
   const [request, doFetch] = useFilms();

   useEffect(() => {
      doFetch();
   }, []);

   return (
      <Column>
         <FilmsList films={request.data?.results} />
      </Column>
   );
};

export default hot(module)(Main);
