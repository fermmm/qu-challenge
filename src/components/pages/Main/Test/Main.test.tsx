import { mount } from 'enzyme';
import React from 'react';
import Main from '../Main';

describe('<Main />', () => {
   test('Renders correctly', () => {
      const component = mount(<Main />);
      expect(component.html()).toMatchSnapshot();
      component.unmount();
   });
});
