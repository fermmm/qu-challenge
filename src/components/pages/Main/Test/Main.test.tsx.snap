// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Renders correctly 1`] = `"<div class=\\"Mainstyles__MainContainer-ipmq2m-0 kZAigZ\\"><div class=\\"FilmListstyles__FilmListContainer-sc-1g4vwsu-0 bKKCvi\\"><div class=\\"FilmListstyles__SortControlsContainer-sc-1g4vwsu-1 hpWsaf\\">Sort by:<button class=\\"FilmListstyles__SortButton-sc-1g4vwsu-2 hVlahb\\">Title</button><button class=\\"FilmListstyles__SortButton-sc-1g4vwsu-2 hVlahb\\">Release Date</button></div></div></div>"`;
